### PR TITLE
[1.5] Don't double quote when conditions #4526

### PR DESCRIPTION
**Pull request by hemmop**

### CODE REVIEW

File: playbooks/adhoc/create_pv/create_pv.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@
   pre_tasks:
   - fail:
       msg: "This playbook requires {{item}} to be set."
-    when: "{{ item }} is not defined or {{ item }} == ''"
+    when: {{ item }} is not defined or {{ item }} == ''
```
Comment (sdodson) [inline playbooks/adhoc/create_pv/create_pv.yaml]: This is different from how it was fixed on master. Need to get rid of jinja delimiters here too.

File: playbooks/aws/openshift-cluster/terminate.yml [hunk 2]
```diff
@@ -40,7 +40,7 @@
         host-type: "{{ hostvars[item]['ec2_tag_host-type'] }}"
         sub_host_type: "{{ hostvars[item]['ec2_tag_sub-host-type'] }}"
     with_items: "{{ groups.oo_hosts_to_terminate }}"
-    when: "'oo_hosts_to_terminate' in groups"
+    when: 'oo_hosts_to_terminate' in groups
```
Comment (sdodson) [inline playbooks/aws/openshift-cluster/terminate.yml]: If it starts with a single quote I believe it needs to maintain double quotes.

**Comment by sdodson** [comment]
Travis should hi-light most of the problems too, see those test results https://travis-ci.org/openshift/openshift-ansible/builds/245370719?utm_source=github_status&utm_medium=notification

File: playbooks/adhoc/create_pv/create_pv.yaml (hunk 1 above)
Comment (hemmop) [inline playbooks/adhoc/create_pv/create_pv.yaml]: Fixed in https://github.com/openshift/openshift-ansible/pull/4527/commits/8247c7a5bad8bb34148071fbc820b21bb09d82e8

File: playbooks/aws/openshift-cluster/terminate.yml (hunk 2 above)
Comment (hemmop) [inline playbooks/aws/openshift-cluster/terminate.yml]: Fixed in https://github.com/openshift/openshift-ansible/pull/4527/commits/4f04379fa47bc451e145d86197179bf240a085f2

File: playbooks/common/openshift-cluster/upgrades/v3_3/upgrade.yml [hunk 3]
```diff
@@ -39,8 +39,8 @@
                                                     | union(groups['oo_etcd_to_config'] | default([])))
                                                 | oo_collect('openshift.common.hostname') | default([]) | join (',')
                                                 }}"
-    when: "{{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
-            openshift_generate_no_proxy_hosts | default(True) | bool }}"
+    when: {{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
```
Comment (mtnbikenc) [inline playbooks/common/openshift-cluster/upgrades/v3_3/upgrade.yml]: Need to remove Jinja delimeters as well.

File: playbooks/common/openshift-cluster/upgrades/v3_3/upgrade_control_plane.yml [hunk 4]
```diff
@@ -47,8 +47,8 @@
                                                     | union(groups['oo_etcd_to_config'] | default([])))
                                                 | oo_collect('openshift.common.hostname') | default([]) | join (',')
                                                 }}"
-    when: "{{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
-            openshift_generate_no_proxy_hosts | default(True) | bool }}"
+    when: {{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
```
Comment (mtnbikenc) [inline playbooks/common/openshift-cluster/upgrades/v3_3/upgrade_control_plane.yml]: Jinja delimiters

File: playbooks/common/openshift-cluster/upgrades/v3_3/upgrade_nodes.yml [hunk 5]
```diff
@@ -40,8 +40,8 @@
                                                     | union(groups['oo_etcd_to_config'] | default([])))
                                                 | oo_collect('openshift.common.hostname') | default([]) | join (',')
                                                 }}"
-    when: "{{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
-            openshift_generate_no_proxy_hosts | default(True) | bool }}"
+    when: {{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
```
Comment (mtnbikenc) [inline playbooks/common/openshift-cluster/upgrades/v3_3/upgrade_nodes.yml]: Jinja delimiters

File: playbooks/common/openshift-cluster/upgrades/v3_4/upgrade.yml [hunk 6]
```diff
@@ -39,8 +39,8 @@
                                                     | union(groups['oo_etcd_to_config'] | default([])))
                                                 | oo_collect('openshift.common.hostname') | default([]) | join (',')
                                                 }}"
-    when: "{{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
-            openshift_generate_no_proxy_hosts | default(True) | bool }}"
+    when: {{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
```
Comment (mtnbikenc) [inline playbooks/common/openshift-cluster/upgrades/v3_4/upgrade.yml]: Jinja delimiters

File: playbooks/common/openshift-cluster/upgrades/v3_4/upgrade_control_plane.yml [hunk 7]
```diff
@@ -47,8 +47,8 @@
                                                     | union(groups['oo_etcd_to_config'] | default([])))
                                                 | oo_collect('openshift.common.hostname') | default([]) | join (',')
                                                 }}"
-    when: "{{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
-            openshift_generate_no_proxy_hosts | default(True) | bool }}"
+    when: {{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
```
Comment (mtnbikenc) [inline playbooks/common/openshift-cluster/upgrades/v3_4/upgrade_control_plane.yml]: Jinja delimiters

File: playbooks/common/openshift-cluster/upgrades/v3_4/upgrade_nodes.yml [hunk 8]
```diff
@@ -40,8 +40,8 @@
                                                     | union(groups['oo_etcd_to_config'] | default([])))
                                                 | oo_collect('openshift.common.hostname') | default([]) | join (',')
                                                 }}"
-    when: "{{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
-            openshift_generate_no_proxy_hosts | default(True) | bool }}"
+    when: {{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
```
Comment (mtnbikenc) [inline playbooks/common/openshift-cluster/upgrades/v3_4/upgrade_nodes.yml]: Jinja delimiters

File: playbooks/common/openshift-cluster/upgrades/v3_5/upgrade.yml [hunk 9]
```diff
@@ -39,8 +39,8 @@
                                                     | union(groups['oo_etcd_to_config'] | default([])))
                                                 | oo_collect('openshift.common.hostname') | default([]) | join (',')
                                                 }}"
-    when: "{{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
-            openshift_generate_no_proxy_hosts | default(True) | bool }}"
+    when: {{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
```
Comment (mtnbikenc) [inline playbooks/common/openshift-cluster/upgrades/v3_5/upgrade.yml]: Jinja delimiters

File: playbooks/common/openshift-cluster/upgrades/v3_5/upgrade_control_plane.yml [hunk 10]
```diff
@@ -47,8 +47,8 @@
                                                     | union(groups['oo_etcd_to_config'] | default([])))
                                                 | oo_collect('openshift.common.hostname') | default([]) | join (',')
                                                 }}"
-    when: "{{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
-            openshift_generate_no_proxy_hosts | default(True) | bool }}"
+    when: {{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
```
Comment (mtnbikenc) [inline playbooks/common/openshift-cluster/upgrades/v3_5/upgrade_control_plane.yml]: Jinja delimiters

File: playbooks/common/openshift-cluster/upgrades/v3_5/upgrade_nodes.yml [hunk 11]
```diff
@@ -40,8 +40,8 @@
                                                     | union(groups['oo_etcd_to_config'] | default([])))
                                                 | oo_collect('openshift.common.hostname') | default([]) | join (',')
                                                 }}"
-    when: "{{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
-            openshift_generate_no_proxy_hosts | default(True) | bool }}"
+    when: {{ (openshift_http_proxy is defined or openshift_https_proxy is defined) and
```
Comment (mtnbikenc) [inline playbooks/common/openshift-cluster/upgrades/v3_5/upgrade_nodes.yml]: Jinja delimiters

File: roles/openshift_certificate_expiry/filter_plugins/oo_cert_expiry.py [hunk 12]
```diff
@@ -35,7 +35,7 @@ def oo_cert_expiry_results_to_json(hostvars, play_hosts):
     become: no
     run_once: yes
     delegate_to: localhost
-    when: "{{ openshift_certificate_expiry_save_json_results|bool }}"
+    when: {{ openshift_certificate_expiry_save_json_results|bool }}
```
Comment (mtnbikenc) [inline roles/openshift_certificate_expiry/filter_plugins/oo_cert_expiry.py]: Jinja delimiters

**Comment by mtnbikenc** [comment]
@hemmop A lot of this has been fixed.  This PR should be rebased and squashed to see what is left.

**Comment by hemmop** [comment]
@mtnbikenc Rebased and squashed

File: roles/openshift_cloud_provider/tasks/openstack.yml [hunk 13]
```diff
@@ -7,4 +7,4 @@
   template:
     dest: "{{ openshift.common.config_base }}/cloudprovider/openstack.conf"
     src: openstack.conf.j2
-  when: "openshift_cloudprovider_openstack_auth_url is defined and openshift_cloudprovider_openstack_username is defined and openshift_cloudprovider_openstack_password is defined and (openshift_cloudprovider_openstack_tenant_id is defined or openshift_cloudprovider_openstack_tenant_name is defined)"
+  when: openshift_cloudprovider_openstack_auth_url is defined and openshift_cloudprovider_openstack_username is defined and openshift_cloudprovider_openstack_password is defined and (openshift_cloudprovider_openstack_tenant_id is defined or openshift_cloudprovider_openstack_tenant_name is defined)
```
Comment (hemmop) [inline roles/openshift_cloud_provider/tasks/openstack.yml]: This line will benefit from a rewrite to
````
when:
 - ...
 - ...
````

syntax

File: roles/openshift_node/tasks/main.yml [hunk 14]
```diff
@@ -121,7 +121,7 @@
     - regex: '^AWS_SECRET_ACCESS_KEY='
       line: "AWS_SECRET_ACCESS_KEY={{ openshift_cloudprovider_aws_secret_key | default('') }}"
   no_log: True
-  when: "openshift_cloudprovider_kind is defined and openshift_cloudprovider_kind == 'aws' and openshift_cloudprovider_aws_access_key is defined and openshift_cloudprovider_aws_secret_key is defined"
+  when: openshift_cloudprovider_kind is defined and openshift_cloudprovider_kind == 'aws' and openshift_cloudprovider_aws_access_key is defined and openshift_cloudprovider_aws_secret_key is defined
```
Comment (hemmop) [inline roles/openshift_node/tasks/main.yml]: Here's a suitable place for a `when` rewrite as well.

**Comment by mtnbikenc** [comment]
@hemmop This PR looks good.  Fine by me if you'd like to re-write those when conditions in list form.

**Comment by sdodson** [comment]
I think we still don't work on branches other than master.

**Comment by mtnbikenc** [comment]
@sdodson, If we don't have CI working on older branches do we merge manually?

**Comment by sdodson** [comment]
Yeah, i'll do that now. I wasn't sure if this was ready to go or not.

**Pull request merged**